tests/regression/test_mesh_vertexonly.py

```diff
@@ -0,0 +1,333 @@
+from firedrake import *
+import pytest
+import numpy as np
+from mpi4py import MPI
+
+# Utility Functions
+
+
+def cell_midpoints(m):
+    """Get the coordinates of the midpoints of every cell in mesh `m`.
+
+    :param m: The mesh to generate cell midpoints for.
+
+    :returns: A tuple of numpy arrays `(midpoints, local_midpoints)` where
+    `midpoints` are the midpoints for the entire mesh even if the mesh is
+    distributed and `local_midpoints` are the midpoints of only the
+    rank-local non-ghost cells."""
+    if isinstance(m.topology, mesh.ExtrudedMeshTopology):
+        raise NotImplementedError("Extruded meshes are not supported")
+    m.init()
+    V = VectorFunctionSpace(m, "DG", 0)
+    f = Function(V).interpolate(m.coordinates)
+    # since mesh may be distributed, the number of cells on the MPI rank
+    # may not be the same on all ranks (note we exclude ghost cells
+    # hence using num_cells_local = m.cell_set.size). Below local means
+    # MPI rank local.
+    num_cells_local = m.cell_set.size
+    num_cells = MPI.COMM_WORLD.allreduce(num_cells_local, op=MPI.SUM)
+    local_midpoints = f.dat.data_ro
+    local_midpoints_size = np.array(local_midpoints.size)
+    local_midpoints_sizes = np.empty(MPI.COMM_WORLD.size, dtype=int)
+    MPI.COMM_WORLD.Allgatherv(local_midpoints_size, local_midpoints_sizes)
+    midpoints = np.empty((num_cells, m.cell_dimension()), dtype=float)
+    MPI.COMM_WORLD.Allgatherv(local_midpoints, (midpoints, local_midpoints_sizes))
+    assert len(np.unique(midpoints, axis=0)) == len(midpoints)
+    return midpoints, local_midpoints
+
+
+"""Parent meshes used in tests"""
```

Comment:
Don't use strings for comments. If you want comments use `#`, if you want to provide a docstring for a variable, you need to do:

```python
my_var = ..
"""Docstring goes here"""
```

Reply:
Changed in 6b2e656b708166ef6ae8d3a03b95bb021bdc3acb . Thanks